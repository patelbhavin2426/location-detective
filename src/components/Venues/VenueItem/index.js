import VenueItem from './venueitem';

export {
    VenueItem
}
