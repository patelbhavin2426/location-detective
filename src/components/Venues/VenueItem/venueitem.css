.tile {
    margin: 10px 0;
    width: 400px;
    padding: 20px;
}

.tile a {
    text-decoration: none;
}

.tile img {
    margin-bottom: 8px;
}

.fas {
    vertical-align: top;
}

.fa.fa-thumbs-up {
    color: mediumseagreen;
}
.fa.fa-thumbs-down {
    color: orange;
}
