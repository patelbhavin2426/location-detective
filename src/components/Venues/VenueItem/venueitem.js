import React, {Component} from 'react'
import './venueitem.css'

function VenueItem(props)  {

    const {venue : { name, verified, location }, id } = props.item;
    const verfifiedClassName = verified?"fa fa-thumbs-up" : "fa fa-thumbs-down";

    return (
        <li data-testid={'venue' + id } className="tile bg-light">
            <div className='location-name'><span className="name">{name}</span> <span className={verfifiedClassName}> </span></div>
            <div>{location.address}</div>
        </li>);

}

export default  VenueItem;
