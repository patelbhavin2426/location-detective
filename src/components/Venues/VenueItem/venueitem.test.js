import React from 'react';
import renderer from 'react-test-renderer';
import VenueItem from './venueitem';
import { mount } from "enzyme";

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe("Venue Item :: ", () => {

    it("should render without throwing an error" , () => {

        let venueList = [
            { venue: { name : 'item one', verified: true, location: {address : 'address one'}} ,  id :'1' },
        ];

        let component = renderer.create(
            <VenueItem item={venueList[0]} />
        );

        // snapshot
        expect(component).toMatchSnapshot();
        expect(component).not.toBe(null);
    });

    it("should render one item", () => {

        let venueList = [
            { venue: { name : 'item two', verified: true, location: {address : 'El Paso venue2'}} , id :'22' }
        ];

        // Act
        let component = mount(
            <VenueItem item={venueList[0]} />
        );
        expect((component).prop('venues')).not.toEqual(null);

        // class check
        expect(component.find('li').first().text()).toEqual('item two  El Paso venue2');

    });

});
