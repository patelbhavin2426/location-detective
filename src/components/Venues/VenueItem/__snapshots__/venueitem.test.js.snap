// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Venue Item ::  should render without throwing an error 1`] = `
<li
  className="tile bg-light"
  data-testid="venue1"
>
  <div
    className="location-name"
  >
    <span
      className="name"
    >
      item one
    </span>
     
    <span
      className="fa fa-thumbs-up"
    >
       
    </span>
  </div>
  <div>
    address one
  </div>
</li>
`;
